mods/ra/maps/allies-10a/allies10a.lua

```diff
@@ -0,0 +1,143 @@
+--[[
+   Copyright 2007-2021 The OpenRA Developers (see AUTHORS)
+   This file is part of OpenRA, which is free software. It is made
+   available to you under the terms of the GNU General Public License
+   as published by the Free Software Foundation, either version 3 of
+   the License, or (at your option) any later version. For more
+   information, see COPYING.
+]]
+StartingUnits = { "mcv", "2tnk", "2tnk", "2tnk", "2tnk" }
+MammothWays = { MammothWay1.Location, MammothWay2.Location, MammothWay3.Location, MammothWay4.Location, ParaLZ5.Location }
+AmbushTeam = { "3tnk", "v2rl", "e2", "e2", "e4" }
+
+StartTimer = false
+TimerColor = Player.GetPlayer("USSR").Color
+TimerTicks = DateTime.Minutes(60)
+ticked = TimerTicks
+
+MissionStart = function()
+	Utils.Do(USSR.GetGroundAttackers(), function(unit)
+		Trigger.OnDamaged(unit, function() IdleHunt(unit) end)
+	end)
+
+	Reinforcements.Reinforce(Greece, StartingUnits, { MCVEntry.Location, MCVStop.Location })
+
+	PatrolMammoth.Patrol(MammothWays, true, 20)
+
+	Trigger.OnKilled(CommandCenter, function()
+		USSR.MarkCompletedObjective(HoldOut)
+	end)
+
+	Trigger.OnEnteredProximityTrigger(FCom.CenterPosition, WDist.FromCells(15), function(actor, id)
+		if actor.Owner == Greece then
+			Trigger.RemoveProximityTrigger(id)
+			LaunchMissiles()
+		end
+	end)
+
+	Trigger.OnEnteredProximityTrigger(OreAmbushTrigger.CenterPosition, WDist.FromCells(5), function(actor, id)
+		if actor.Owner == Greece and actor.Type == "harv" then
+			Trigger.RemoveProximityTrigger(id)
+			local ambush = Reinforcements.Reinforce(USSR, AmbushTeam, { OreAmbushEntry.Location})
+			Utils.Do(ambush, IdleHunt)
+		end
+	end)
+end
+
+LaunchMissiles = function()
```

Comment:
We can use something like this to play the FMV:
```lua
LaunchMissiles = function()
	Media.PlaySoundNotification(Greece, "AlertBleep")
	Media.DisplayMessage("INCOMING TRANSMISSION", "LANDCOM 16")

	Trigger.AfterDelay(DateTime.Seconds(1), function()
		fmvStart = DateTime.GameTime
		Media.PlayMovieFullscreen("ally10b.vqa", function()
			-- Completing immediately indicates that the FMV is not available
			-- Fall back to a text message
			if DateTime.GameTime == DateTime.GameTime then
				Media.PlaySpeechNotification(Greece, "AbombLaunchDetected")
				Media.DisplayMessage("Commander, we're tracking four missiles. They must be deactivated! We are scrambling a team to assult the missile control center. Clear the way and capture the enemy command center. Hurry!", "LANDCOM 16")
			end
		end)
	end)

	Trigger.AfterDelay(DateTime.Seconds(4), function()
		CaptureFCom = Greece.AddObjective("Capture the enemy Command Center. Hurry!")
		StartTimer = true
		Media.PlaySpeechNotification(Greece, "TimerStarted")
		Media.DisplayMessage("Commander, we're tracking four missiles. They must be deactivated! We are scrambling a team to assult the missile control center. Clear the way and capture the enemy command center. Hurry!", "LANDCOM 16")
		Greece.MarkCompletedObjective(ApproachBase)
	end)

	Trigger.OnCapture(CommandCenter, function()
		Greece.MarkCompletedObjective(CaptureFCom)
	end)

	Paradrop()
	Trigger.AfterDelay(DateTime.Minutes(2), SendParabombs)
end
```

Reply:
What do you think about using a camera to reveal the silos (the player is already nearby so this isn't revealing much they won't soon already know) and script the launch a couple of seconds before showing the FMV/message?